File: app/schemas/history.py
from typing import List, Optional
from pydantic import BaseModel

class FrontendHistoryItem(BaseModel):
    timestamp: str
    local_id: str
    status: str
    observacao: str
    responsavel: str
    contato: str
    mensagem_original: Optional[str] = None
    message_id: Optional[str] = None

class HistoryListResponse(BaseModel):
    items: List[FrontendHistoryItem]
    total: int

File: web/src/types/history.ts
export interface HistoryItem {
  timestamp: string;
  local_id: string;
  status: string;
  observacao: string;
  responsavel: string;
  contato: string;
  mensagem_original?: string;
  message_id?: string;
}

export interface HistoryResponse {
  items: HistoryItem[];
  total: number;
}
